packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1773,21 +1794,118 @@ describe('TransactionController', () => {
         'ApprovalController:addRequest',
         {
           expectsResult: true,
-          id: 'mocked1',
+          id: '1234',
           origin: 'metamask',
-          requestData: { txId: 'mocked1' },
+          requestData: { txId: '1234' },
           type: 'transaction',
         },
         false,
       );
     });
 
     it('does not create any approval when there is no unapproved transaction', async () => {
-      const controller = newController();
-      controller.initApprovals();
-
+      newController();
       expect(delayMessengerMock.call).not.toHaveBeenCalled();
     });
+
+    it('updates gas values of unapproved transactions', async () => {
+      const mockTransactionMeta = {
+        from: ACCOUNT_MOCK,
+        chainId: toHex(5),
+        status: TransactionStatus.unapproved,
+        txParams: {
+          from: ACCOUNT_MOCK,
+          to: ACCOUNT_2_MOCK,
+        },
+      };
+      const mockGasPrice = '0x1';
+      const mockedTransactions = [
+        {
+          id: '123',
+          ...mockTransactionMeta,
+          history: [{ ...mockTransactionMeta, id: '123' }],
+        },
+        {
+          id: '1234',
+          ...mockTransactionMeta,
+          history: [{ ...mockTransactionMeta, id: '1234' }],
+        },
+      ];
+
+      const mockedControllerState = {
+        transactions: mockedTransactions,
+        methodData: {},
+        lastFetchedBlockNumbers: {},
+      };
+
+      updateGasFeesMock.mockImplementation(({ txMeta }) => {
+        // Assume this is a sample update
+        txMeta.txParams.gasPrice = mockGasPrice;
+        return Promise.resolve();
+      });
+
+      const controller = newController({
+        state: mockedControllerState as any,
+      });
+
+      await new Promise((r) => setTimeout(r, 0));
```

Comment:
These are just to make controller constructor async functions to be resolve.

Reply:
Should we put this in a `flushPromises` method for better readability?

Reply:
Good call, I put this in https://github.com/MetaMask/core/pull/1916/commits/524bb0ffe7881ec08d99780d540c712c3eb64ebf